ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -244,16 +249,30 @@ def start_span(
 
     def inject(self, span_context, format, carrier):
         # pylint: disable=redefined-builtin
-        logger.warning(
-            "Calling unimplemented method inject() on class %s",
-            self.__class__.__name__,
-        )
-        # TODO: Implement.
+        # This implementation does not perform the injecting by itself but
+        # uses the configured propagators in opentelemetry.propagators.
+        # TODO: Support Format.BINARY once it is supported in
+        # opentelemetry-python.
+        if format not in self._supported_formats:
+            raise opentracing.UnsupportedFormatException
+
+        propagator = propagators.get_global_httptextformat()
+        propagator.inject(span_context.unwrap(), dict.__setitem__, carrier)
```

Comment:
```suggestion
        propagator.inject(span_context.unwrap(), type(carrier).__setitem__, carrier)
```

I'm not sure about that though.

Reply:
That could be better because is more generic. Thanks!